working/0093 - Gradual Language Change Migration/0094 - Per Library Language Version Selection/strawman.md

```diff
@@ -38,7 +38,17 @@ A package selects the _language version level_ it expects using its `pubspec.yam
 A Dart SDK tool must reject a any version higher than what the SDK supports. A Dart SDK may also have a lowest version that it
 supports being backwards compatible with.
 
-We will allow an individual library to select a lower language version level than the one selected for its package, for the case where that library has not yet been migrated to the newer semantics. 
+**Bob: This is unclear, since the SDK constraint specifies a *range* of
+versions. I think what this should do is that the *minimum* version of the
+constraint determines the default language level all code in the package can
+assume. If a Dart SDK tool's version is not within the constraint at all, that's
+an error.**
+
+We will allow an individual library to select a lower language version level than the one selected for its package, for the case where that library has not yet been migrated to the newer semantics.
+
+**Bob: Should tools make it an error for a library to select a *higher* version?
+Doing so will always cause problems because users of older SDKs could end up
+with this library but not be able to use it.**
```

Comment:
I believe line 119 confirms this. Maybe it would be nice to move that sentence up here.